networking_aci/plugins/ml2/drivers/mech_aci/driver.py

```diff
@@ -200,16 +201,37 @@ def _get_clearable_phys_doms(self, network, local_segment, host_config):
         #       get host
         session = db_api.get_reader_session()
         with session.begin():
-            other_bindings = (session.query(models.PortBindingLevel.host, models.PortBindingLevel.segment_id)
+            other_bindings = (session.query(models.PortBindingLevel.host, models.PortBindingLevel.segment_id,
+                                            models.PortBinding.profile)
                                      .join(segment_model.NetworkSegment,
                                            segment_model.NetworkSegment.id == models.PortBindingLevel.segment_id)
+                                     .join(models.PortBinding,
+                                           models.PortBindingLevel.port_id == models.PortBinding.port_id)
                                      .filter(segment_model.NetworkSegment.network_id == network['id'],
                                              models.PortBindingLevel.level == 1,
                                              models.PortBindingLevel.segment_id != local_segment['id'])
                                      .distinct())
 
         for other_binding in other_bindings:
-            _, other_binding_host_config = self._host_or_host_group(other_binding.host)
+            host = other_binding.host
+
+            # try to get alternate host from binding profile
+            binding_profile = None
+            try:
+                binding_profile = json.loads(other_binding.profile)
+                if binding_profile:
+                    switch = CiscoACIMechanismDriver.switch_from_local_link(binding_profile)
+                    if switch:
+                        host = switch
+            except json.decoder.JSONDecodeError:
+                # don't act on broken binding profile
+                pass
+
+            _, other_binding_host_config = self._host_or_host_group(host)
+            if not other_binding_host_config:
+                LOG.warning("No config available / invalid binding host %s, segment %s with profile %s in network %s",
+                            other_binding.host, other_binding.segment_id, other_binding.profile, network['id'])
```

Comment:
This log message needs to use `host` instead of `other_binding.host`, too, doesn't it?

Reply:
I originally intended here to show binding host, segemnt id and binding profile from the database to show to the user on what basis the driver decided to print this message. Using host would also be an option, then we would only show the derived data... Hmm. Doesn't sound like a bad idea.